src/main/java/com/example/bookstore/service/impl/OrderItemServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package com.example.bookstore.service.impl;
+
+import com.example.bookstore.dto.order.OrderItemResponseDto;
+import com.example.bookstore.exception.EntityNotFoundException;
+import com.example.bookstore.mapper.OrderItemMapper;
+import com.example.bookstore.model.Order;
+import com.example.bookstore.repository.OrderItemRepository;
+import com.example.bookstore.repository.OrderRepository;
+import com.example.bookstore.service.OrderItemService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderItemServiceImpl implements OrderItemService {
+    private final OrderItemRepository orderItemRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    public List<OrderItemResponseDto> getAllOrderItemsFromOrder(Long orderId, Pageable pageable) {
+        return orderItemRepository.findAllByOrderId(orderId, pageable).stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemResponseDto getOrderItemFromOrder(Long userId,
+                                                  Long orderId, Long orderItemId) {
+        Order order = orderRepository.findByIdAndUserId(orderId, userId).orElseThrow(
+                () -> new EntityNotFoundException("Cannot find order by id: " + orderId)
+        );
+        return orderItemMapper.toDto(order.getOrderItems().stream()
+                .filter(orderItem -> orderItem.getId().equals(orderItemId))
+                .findFirst()
+                .orElseThrow(
+                        () -> new EntityNotFoundException("Cannot find orderItem by id: "
+                                + orderItemId)));
```

Comment:
GJ, but you can simplify it. Try to create method to find order item by query

Reply:
yes, can be simplified with only one `repo` call